### PR TITLE
Remove userAllow instances.

### LINKED ISSUE #1528: remove badger.userAllow
We do random bookeeping for this thing. We only use it once, in that place using 

```javascript
if (badger.storage.getAction(domain) === constants.USER_ALLOW){
```
would be more appropriate

```bash
➜  js git:(master) ✗ grep -ri userallow
storage.js:   * userAllow list if needed
storage.js:    var index = badger.userAllow.indexOf(domain);
storage.js:      badger.userAllow.splice(index, 1);
storage.js:      badger.userAllow.push(domain);
storage.js:   * Remove user set action from a domain and remove it from userAllow
storage.js:    var index = badger.userAllow.indexOf(domain);
storage.js:      badger.userAllow.splice(index, 1);
background.js:  this.userAllow = [];
background.js:      badger.initializeUserAllowList();
background.js:  initializeUserAllowList: function() {
background.js:        this.userAllow.push(domain);
webrequest.js:    socialWidgetsToReplace[socialWidgetName] = !(badger.userAllow.indexOf(socialwidget.domain) > -1);

```

### PR DESCRIPTION
**Pull request by roberthluo**
Fixes #1528. 

Just removes all instances of userAllow. 

Update: Added new tests to test user override of default action in domain.

This is my first contribution so please let me know if I've done anything incorrectly or if you like for me to make any changes.

### CODE REVIEW

**Comment by roberthluo** [comment]
Not sure why the Travis CI build could not complete. It says 

>  This job ran on our Trusty environment, which has been updated on Tuesday, December 12th, 2017. Read all about this on our blog and take note that you can add group: deprecated-2017Q4 in your .travis.yml file to use the previous image version. 

Do I need to do anything?

**Comment by ghostwords** [comment]
Do you mean [this failed job](https://travis-ci.org/EFForg/privacybadger/jobs/317854660)? I think it just failed to fetch Python 3.4 because of a networking hiccup. Doesn't seem related to your PR; all the other jobs passed.

**Comment by ghostwords** [comment]
Thank you for the contribution! I'll review it in the next few days. Don't worry about the failed job, Travis just flakes out sometimes.

**Comment by ghostwords** [comment]
Unfortunately, I was sick this week and didn't get to fully review your PR. I do have a couple of preliminary questions. I plan on finishing the review after the holidays in the new year.

- Was the problem with the original optimization in #938 that it did too much, the only thing that really needed to change was to avoid calling `getAllDomainsByPresumedAction` and instead look up actions for just the domains in `SocialWidgetList`?
- Since the unit tests were added to support `userAllow` in #938, are they still necessary?

**Comment by roberthluo** [comment]
Yes, I think the problem with the original optimization in #938 that it did too much, the only thing that really needed to change was to avoid calling `getAllDomainsByPresumedAction` and instead look up actions for just the domains in `SocialWidgetList` instead of generating an `userAllow` list .

The unit tests added to support `userAllow` in #938 don't exist anymore (they were removed), but they were different. In #938 they checked if the list existed by looking at the elements of the list, but mine uses `getAction` to see the associated action related to the `domain`.

File: src/js/webrequest.js [hunk 1]
```diff
@@ -548,9 +548,9 @@ function getSocialWidgetBlockList() {
 
   window.SocialWidgetList.forEach(function(socialwidget) {
     var socialWidgetName = socialwidget.name;
-
-    // Only replace social widgets that the user has not manually allowed
```
Comment (ghostwords) [inline src/js/webrequest.js]: Should keep this comment, I think.

File: src/js/webrequest.js [hunk 2]
```diff
@@ -548,9 +548,9 @@ function getSocialWidgetBlockList() {
...
-    socialWidgetsToReplace[socialWidgetName] = !(badger.userAllow.indexOf(socialwidget.domain) > -1);
+    socialWidgetsToReplace[socialWidgetName] = !(
```
Comment (ghostwords) [inline src/js/webrequest.js]: We should simplify the comparison by moving the `!` to the equality test itself:
```javascript
    socialWidgetsToReplace[socialWidgetName] = (
      badger.storage.getAction(socialwidget.domain) != constants.USER_ALLOW
    );
```

File: src/tests/tests/storage.js [hunk 3]
```diff
@@ -26,15 +26,15 @@
...
     badger.saveAction("block", "pbtest.org");
-    assert.ok(badger.userAllow.indexOf('pbtest.org') <= -1);
+    assert.ok(badger.storage.getAction("pbtest.org") === constants.USER_BLOCK);
     badger.saveAction("allow", "pbtest.org");
-    assert.ok(badger.userAllow.indexOf('pbtest.org') > -1);
-    storage.revertUserAction("pbtest.org");
-    assert.ok(badger.userAllow.indexOf('pbtest.org') <= -1);
+    assert.ok(badger.storage.getAction("pbtest.org") === constants.USER_ALLOW);
+    badger.storage.revertUserAction("pbtest.org");
+    assert.ok(badger.storage.getAction("pbtest.org") === "noaction");
```
Comment (ghostwords) [inline src/tests/tests/storage.js]: `"noaction"` should be `constants.NO_TRACKING`.

File: src/tests/tests/storage.js [hunk 4]
```diff
@@ -26,15 +26,15 @@
...
   });
-
+
```
Comment (ghostwords) [inline src/tests/tests/storage.js]: Extra trailing space.

**Comment by ghostwords** [comment]
Looks good, merged. Thank you!

**Pull request merged**